package.json

```diff
@@ -13,44 +13,44 @@
   },
   "private": false,
   "dependencies": {
-    "@angular/animations": "^5.2.2",
-    "@angular/cdk": "^5.1.0",
-    "@angular/common": "^5.2.2",
-    "@angular/compiler": "^5.2.2",
-    "@angular/core": "^5.2.2",
-    "@angular/forms": "^5.2.2",
-    "@angular/http": "^5.2.2",
-    "@angular/material": "^5.1.0",
-    "@angular/platform-browser": "^5.2.2",
-    "@angular/platform-browser-dynamic": "^5.2.2",
-    "@angular/platform-server": "^5.2.2",
-    "@angular/router": "^5.2.2",
-    "core-js": "^2.5.1",
-    "hammerjs": "^2.0.8",
+    "@angular/animations": "*",
+    "@angular/cdk": "*",
```

Comment:
Material & CDK `6.0.0-beta.5` is out now which I think should contain your change. You probably want to use that instead of the `*-builds` repos. For the rest of the Angular stuff you can use `6.0.0-rc.0`.

Reply:
@mmalerba  Thanks!

Now I get 

```
ERROR in node_modules/@angular/common/src/location/location.d.ts(1,10): error TS2305: Module '"/Users/.../gitreps/JDNConvertibleCalendarDateAdapter/node_modules/rxjs/Rx"' has no exported member 'SubscriptionLike'.
src/app/modules/jdnconvertiblecalendardateadapter/adapter/index.ts(2,57): error TS2305: Module '"/Users/.../gitreps/JDNConvertibleCalendarDateAdapter/node_modules/@angular/material/material"' has no exported member 'MAT_DATE_LOCALE_PROVIDER'.
```

For error 1 I tried the solution mentioned here  <https://stackoverflow.com/questions/49446163/typescript-how-to-resolve-rxjs-rx-has-no-exported-member-subscriptionlike> but then at runtime it did not work.

For error 2: Was  `MAT_DATE_LOCALE_PROVIDER` removed from `angular/material`?

Reply:
We have `rxjs 6.0.0-beta.1` in our `package.json`, you might need to update that too. Yes, we removed `MAT_DATE_LOCALE_PROVIDER` in favor of the new scoped injectables (see https://github.com/angular/material2/pull/10507). Theoretically this means you can just remove the reference to it

Reply:
Do I need to update the @types libs?

I get these errors at build time

```
ERROR in node_modules/@angular/material/table/typings/table-data-source.d.ts(130,5): error TS2416: Property 'connect' in type 'MatTableDataSource<T>' is not assignable to the same property in base type 'DataSource<T>'.
  Type '() => BehaviorSubject<T[]>' is not assignable to type '(collectionViewer: CollectionViewer) => Observable<T[]>'.
    Type 'BehaviorSubject<T[]>' is not assignable to type 'Observable<T[]>'.
      Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
node_modules/@angular/material/tree/typings/data-source/flat-data-source.d.ts(80,5): error TS2416: Property 'connect' in type 'MatTreeFlatDataSource<T, F>' is not assignable to the same property in base type 'DataSource<F>'.
  Type '(collectionViewer: CollectionViewer) => Observable<F[]>' is not assignable to type '(collectionViewer: CollectionViewer) => Observable<F[]>'. Two different types with this name exist, but they are unrelated.
    Type 'Observable<F[]>' is not assignable to type 'Observable<F[]>'. Two different types with this name exist, but they are unrelated.
      Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
node_modules/@angular/material/tree/typings/data-source/nested-data-source.d.ts(23,5): error TS2416: Property 'connect' in type 'MatTreeNestedDataSource<T>' is not assignable to the same property in base type 'DataSource<T>'.
  Type '(collectionViewer: CollectionViewer) => Observable<T[]>' is not assignable to type '(collectionViewer: CollectionViewer) => Observable<T[]>'. Two different types with this name exist, but they are unrelated.
    Type 'Observable<T[]>' is not assignable to type 'Observable<T[]>'. Two different types with this name exist, but they are unrelated.
```

Reply:
Hmm I'm not sure why that's happening. Here's what our own `package.json` looked like as of the 6.0.0-beta.5 release: https://github.com/angular/material2/blob/0e37d6cdd73b49923d06add54e0d34bd9d256eda/package.json

Make sure your dependencies are at a high enough version, delete your `package-lock.json` and `node_modules/` and then run `npm i`. *Hopefully* that should make it work. (There may be breakages in your code, since these are new major versions, but it should at least get rid of the errors coming from the Angular Material code).

Reply:
Ok, it works now:-) Thanks!